dusty/command_file.py

```diff
@@ -0,0 +1,129 @@
+    commands += testing_spec['once']
+    return commands
+
+def _lib_install_commands_for_libs(assembled_specs, libs):
+    commands = []
+    for lib in libs:
+        lib_spec = assembled_specs['libs'][lib]
+        install_commands = _lib_install_commands(lib_spec)
+        if install_commands:
+            commands += install_commands
+    return commands
+
+def _lib_install_commands_for_app(app_name, assembled_specs):
+    """ This returns a list of all the commands that will install libraries for a
+    given app """
+    libs = assembled_specs['apps'][app_name]['depends']['libs']
+    return _lib_install_commands_for_libs(assembled_specs, libs)
+
+def _lib_install_commands_for_lib(app_name, assembled_specs):
+    """ This returns a list of all the commands that will install libraries for a
+    given lib """
+    libs = assembled_specs['libs'][app_name]['depends']['libs']
+    return _lib_install_commands_for_libs(assembled_specs, libs)
+
+def lib_install_commands_for_app_or_lib(app_or_lib_name, assembled_specs):
+    if app_or_lib_name in assembled_specs['apps']:
+        return _lib_install_commands_for_app(app_or_lib_name, assembled_specs)
+    return _lib_install_commands_for_lib(app_or_lib_name, assembled_specs)
+
+def _lib_install_commands(lib_spec):
+    """ This returns a single commmand that will install a library in a docker container """
+    if not lib_spec['install']:
+        return []
+    return ["cd {}".format(lib_spec['mount'])] + lib_spec['install']
+
+def dusty_command_file_name(app_or_lib_name, script_name=None, test_name=None):
+    command_file_name = 'dusty_command_file_{}'.format(app_or_lib_name)
+    if script_name:
+        command_file_name = '{}_script_{}'.format(command_file_name, script_name)
+    elif test_name:
+        command_file_name = '{}_test_{}'.format(command_file_name, test_name)
+    return "{}.sh".format(command_file_name)
+
+def _get_command_file_path(app_or_lib_name, assembled_specs, script_name=None, test_name=None):
+    if app_or_lib_name in assembled_specs['apps']:
+        spec = assembled_specs['apps'][app_or_lib_name]
+    else:
+        spec = assembled_specs['libs'][app_or_lib_name]
+    file_name = dusty_command_file_name(app_or_lib_name, script_name=script_name, test_name=test_name)
+    repo = Repo(spec['repo'])
+    return '{}/{}'.format(repo.local_path, file_name)
+
+def make_up_command_files(assembled_specs):
+    for app_name in assembled_specs['apps'].keys():
+        commands = _compile_docker_commands(app_name, assembled_specs)
+        _write_commands_to_file(commands, _get_command_file_path(app_name, assembled_specs))
+        script_specs = assembled_specs['apps'][app_name]['scripts']
+        for script_spec in script_specs:
+            commands = ["cd {}".format(container_code_path(assembled_specs['apps'][app_name]))] + script_spec['command']
+            commands[-1] = '{} $@'.format(commands[-1])
```

Comment:
The $@ is a way to specify arguments passed in, in a bash file.  This will pass all arguments to the final command.